projects/orangecrab/Makefile

```diff
@@ -0,0 +1,78 @@
+##----------------------------------------------------------------------------##
+#   Project Settings                                                           #
+##----------------------------------------------------------------------------##
+
+NAME = Blink
+TOP = topEntity
+BUILDDIR= _build
+PNR_FLAGS= \
+	--85k \
+	--package CSFBGA285 \
+	--lpf orangecrab.pcf \
+
+-include build.cfg
+-include build.cfg.local
+
+##----------------------------------------------------------------------------##
+#   Build Rules                                                                #
+##----------------------------------------------------------------------------##
+
+default: bitstream
+
+upload: ${BUILDDIR}/04-bitstream/${TOP}.bit
+	${PROG} -S $<
+
+hdl: ${BUILDDIR}/02-hdl/${TOP}.v
+
+synth: ${BUILDDIR}/03-net/$(TOP).json
+
+netlist: ${BUILDDIR}/04-bitstream/$(TOP).config
+
+bitstream: ${BUILDDIR}/04-bitstream/$(TOP).bit
+
+${BUILDDIR}/04-bitstream/${TOP}.dfu: ${BUILDDIR}/04-bitstream/${TOP}.bit
+	cp -a $< $@
+	${DFUSUFFIX} -v 1209 -p 5af0 -a $@
+
+${BUILDDIR}/04-bitstream/${TOP}.bit: ${BUILDDIR}/04-bitstream/${TOP}.config
+	${PACK} --compress --freq 38.8 --input $< --bit $@
+
+${BUILDDIR}/04-bitstream/$(TOP).config: ${BUILDDIR}/03-net/$(TOP).json
+	mkdir -p ${BUILDDIR}/04-bitstream
+	${PNR} --json $< --textcfg $@ ${PNR_FLAGS}
+
+${BUILDDIR}/03-net/$(TOP).json: ${BUILDDIR}/02-hdl/${TOP}.v
+	mkdir -p ${BUILDDIR}/03-net ${BUILDDIR}/log
+	$(YOSYS) \
+		-l ${BUILDDIR}/log/synth.log \
+		-p 'read_verilog $(dir $^)/*.v; synth_ecp5 -top $(TOP) -json $@'
+
+${BUILDDIR}/02-hdl/${TOP}.v: src/${NAME}.hs dist-newstyle/packagedb src/*
+	${CABAL} build
+	mkdir -p ${BUILDDIR}/01-clash
+	echo
+	${CABAL} run clash -- \
+		-package-env $(abspath).ghc.environment.* \
+		--verilog \
+		-outputdir ${BUILDDIR}/01-clash \
+		$<
```

Comment:
This invokes
```
cabal run clash -- src/Blink.hs
```
but that is not truly the correct thing for a project with the structure of our starter projects. You should just invoke
```
cabal run clash -- Blink
```
which will pick up the entity from the package built by Cabal.

Passing the source file means you again compile `Blink.hs` from source and only its imports come from pre-compiled stuff. This also means none of the things in `orangecrab.cabal` or `cabal.project` apply to `Blink.hs`, it is compiled with the builtin settings of the Clash compiler. There are situations where the two are simply not equivalent at all.